File: smart_farm/src/api/diseasePest.ts
import { get, http, post } from '@/utils/request'
import type { PageInfo } from '@/api/user'

/**
 * 病虫害基本信息实体，对应 smart_plant 后端 disease_pest 表返回字段。
 * 前端列表、详情弹框和防治措施关联选择都复用该类型，避免字段散落在页面中。
 */
export interface DiseasePest {
  id?: number
  cropTypeId?: number | null
  cropTypeName?: string
  affectedCrops?: string
  name: string
  type?: number
  symptom?: string
  cause?: string
  suitableStage?: string
  occurrencePeriod?: string
  livingHabits?: string
  suitableEnvironment?: string
  transmissionRoute?: string
  coverImage?: string
  imageUrls?: string[]
  status?: number
  sortOrder?: number
  remark?: string
  createTime?: string
  updateTime?: string
}

/**
 * 新增病虫害基本信息时提交给后端的字段。
 * id、createTime、updateTime 等字段由数据库维护，新增表单不提交。
 */
export interface DiseasePestPayload {
  cropTypeId?: number | null
  affectedCrops?: string
  name: string
  type?: number
  symptom?: string
  cause?: string
  suitableStage?: string
  occurrencePeriod?: string
  livingHabits?: string
  suitableEnvironment?: string
  transmissionRoute?: string
  coverImage?: string
  imageUrls?: string[]
  status?: number
  sortOrder?: number
  remark?: string
}

export interface UpdateDiseasePestPayload extends Partial<DiseasePestPayload> {
  id: number
}

export interface DiseasePestListParams {
  name?: string
  cropTypeId?: number
  type?: number
  status?: number
  pageNum: number
  pageSize: number
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

/** 分页查询病虫害基本信息列表，供列表页表格和下拉选择使用。 */
export const listDiseasePests = (params: DiseasePestListParams) => {
  return get<ApiResult<PageInfo<DiseasePest>>>(
    '/disease-pest/list',
    params as unknown as Record<string, unknown>,
  )
}

/** 根据 ID 查询病虫害基本信息详情，供编辑弹框回填。 */
export const getDiseasePestById = (id: number) => {
  return get<ApiResult<DiseasePest>>(`/disease-pest/${id}`)
}

/** 新增病虫害基本信息。 */
export const addDiseasePest = (data: DiseasePestPayload) => {
  return post<ApiResult<DiseasePest>>(
    '/disease-pest/add',
    data as unknown as Record<string, unknown>,
  )
}

/** 编辑病虫害基本信息。 */
export const updateDiseasePest = (data: UpdateDiseasePestPayload) => {
  return http<ApiResult<DiseasePest>>({ url: '/disease-pest', method: 'put', data })
}

/** 删除单条病虫害基本信息，后端会级联删除该病虫害下的防治措施。 */
export const deleteDiseasePest = (id: number) => {
  return http<ApiResult<void>>({ url: `/disease-pest/${id}`, method: 'delete' })
}

/** 批量删除病虫害基本信息。 */
export const batchDeleteDiseasePests = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/disease-pest/batch', method: 'delete', data: ids })
}

/** 启用或停用病虫害基本信息。 */
export const updateDiseasePestStatus = (id: number, status: number) => {
  return http<ApiResult<void>>({
    url: `/disease-pest/${id}/status`,
    method: 'put',
    params: { status },
  })
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/DiseasePest.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;

/**
 * 病虫害基础信息实体。
 *
 * <p>该实体对应数据库中的 disease_pest 表，保存病害、虫害和生理性病害的基础知识。</p>
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class DiseasePest {

    /** 病虫害主键ID。 */
    private Long id;

    /** 关联作物类型ID，对应 crop_type.id，可为空表示通用病虫害。 */
    private Long cropTypeId;

    /** 作物类型名称，列表展示时通过关联 crop_type 表带出。 */
    private String cropTypeName;

    /** 危害作物范围展示文本，用于表达跨多个作物类型的适用范围。 */
    private String affectedCrops;

    /** 病虫害名称。 */
    private String name;

    /** 类型：1病害，2虫害，3生理性病害。 */
    private Integer type;

    /** 症状描述。 */
    private String symptom;

    /** 发生原因。 */
    private String cause;

    /** 易发生阶段，例如苗期、花期、成熟期等。 */
    private String suitableStage;

    /** 主要发生时期或季节，例如春秋季或4月至6月。 */
    private String occurrencePeriod;

    /** 害虫、病原或生理性病害的生活习性。 */
    private String livingHabits;

    /** 适宜发生的温度、湿度、连作等环境条件。 */
    private String suitableEnvironment;

    /** 飞行、土壤、种苗调运等传播或扩散途径。 */
    private String transmissionRoute;

    /** 封面图片地址。 */
    private String coverImage;

    /** 详情页图片地址集合，以 MySQL JSON 数组持久化。 */
    private List<String> imageUrls;

    /** 状态：1启用，0停用。 */
    private Integer status;

    /** 展示顺序，数值越小越靠前。 */
    private Integer sortOrder;

    /** 备注信息，用于记录补充说明。 */
    private String remark;

    /** 创建时间，由数据库默认维护。 */
    private LocalDateTime createTime;

    /** 更新时间，由数据库在更新时自动维护。 */
    private LocalDateTime updateTime;

    /** 防治措施列表，详情查询时由 service 组装。 */
    private List<DiseaseControl> controls;
}
